cli/src/commands/git.rs

```diff
@@ -107,6 +108,49 @@ pub struct GitRemoteRenameArgs {
 #[derive(clap::Args, Clone, Debug)]
 pub struct GitRemoteListArgs {}
 
+/// Create a new Git backed repo.
+#[derive(clap::Args, Clone, Debug)]
+pub struct GitInitArgs {
+    /// The destination directory where the `jj` repo will be created.
+    /// If the directory does not exist, it will be created.
+    /// If no directory is diven, the current directory is used.
+    ///
+    /// By default the `git` repo is under `$destination/.jj/repo/store/git`
+    #[arg(default_value = ".", value_hint = clap::ValueHint::DirPath)]
+    destination: String,
+
+    /// Specifies that the `jj` repo should also be a valid
+    /// `git` repo, allowing the use of both `jj` and `git` commands
+    /// in the same directory.
+    ///
+    /// This is done by placing the backing git repo into a `.git` folder
+    /// in the root of the Jujutsu repo along with the `.jj` folder.
+    ///
+    /// This option is only valid when creating new repos. To
+    /// reuse an existing `.git` directory in an existing git
+    /// repo, see the `--git-repo` param below.
```

Comment:
Ah, good point. While it makes sense to not allow `jj git init --colocated` in existing git repo, it would be nice if we had an easy way to set up colocated repo in that way. `--git-repo=.` isn't useful if the destination path isn't cwd.

Reply:
I am thinking of following in subsequent PRs to add:

 --allow-existing - Use with --colocated to replace --git-repo=.; internally we'll still call init_git_external, but the user won't care.

--autopropagate? - Allow auto import/export to **existing** colocated repos. This will probably need wiring some flags to init_external_git.

If these seem like good ideas, I'll work on them after the initial simple version is merged.

Reply:
I'm not sure if adding flags is better, but this isn't a blocker of this PR. We can decide that later.

(I personally prefer `jj git init` that creates a colocated repo by default, and imports changes from `.git` repo if already exists.)

Reply:
Makes sense. I'll bring this up for discussion on the issue itself.

Thanks.